### ISSUE TITLE
Synchrotron: Error processing replication

### ISSUE THREAD

**Issue by hg-runs** [matrix-org/synapse#6428]
### Description

Found the following exception in the log (not sure if this is a bug at all):

```
2019-11-27 15:45:20,666 - synapse.app.synchrotron - 426 - ERROR - replication-RDATA-events-551 - Error processing replication
Traceback (most recent call last):
  File "/opt/syntest/synapse/env/lib/python3.7/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
    result = g.send(result)
StopIteration: {'$9Ep67Yo6EdH9NpHQaBDkE4Czl_mMf7rqJua7mYkkoAA': {'event_id': '$9Ep67Yo6EdH9NpHQaBDkE4Czl_mMf7rqJua7mYkkoAA', 'internal_metadata': '{"outlier": false, "soft_failed": true, "stream_ordering": 63480}', 'json': '{"auth_events": ["$eqyMb8L3V8Y7spsP40m1uEvQpPbpMJs-aCqKrD2sh4U", "$44vIat4gq6zdBgaaPeWPRZDjuU09wigV-dDKUhh4ffw", "$tKDzaQxlSz5suAUwjtMKsyfXtIOsID5wJFtTiK5LtZM"], "content": {"body": "Thanks for all the help and the learning opportunity tulir and houtworm / Houtworm ", "format": "org.matrix.custom.html", "formatted_body": "Thanks for all the help and the learning opportunity <a href=\\"https://matrix.to/#/@tulir:maunium.net\\">tulir</a> and <a href=\\"https://matrix.to/#/@houtworm:matrix.org\\">houtworm</a> / <a href=\\"https://matrix.to/#/@houtworm:houtworm.im\\">Houtworm</a>", "msgtype": "m.text"}, "depth": 29242, "hashes": {"sha256": "sCWWEXf6gcWmMu38K2GSDAYp3PMxyT09G7vC8JPAhhc"}, "origin": "riot.garwood.io", "origin_server_ts": 1572545417280, "prev_events": ["$7lAEXEHbrSjK4Q8__NKEmPq_7LDbU2wLlJZQE_5PxdU"], "prev_state": [], "room_id": "!mjbDjyNsRXndKLkHIe:matrix.org", "sender": "@clifford:riot.garwood.io", "type": "m.room.message", "signatures": {"riot.garwood.io": {"ed25519:a_fKNq": "hh/SQ77aR/wWRUMI2vQoMTvcP4CRqBIWnWpHK+G7Ukt5jKJFvh9EeHSGVYMlOwlyTZLLfsD9cnlFuiIRN106CQ"}}, "unsigned": {"age": 1565192287}}', 'format_version': 3, 'rejected_reason': 'auth_error', 'redactions': []}}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/syntest/synapse/env/lib/python3.7/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
    result = g.send(result)
StopIteration: {}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/syntest/synapse/env/lib/python3.7/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
    result = g.send(result)
StopIteration: {}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/syntest/synapse/env/lib/python3.7/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
    result = g.send(result)
StopIteration: []

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/syntest/synapse/synapse/app/synchrotron.py", line 384, in process_and_notify
    event = yield self.store.get_event(row.data.event_id)
  File "/opt/syntest/synapse/env/lib/python3.7/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
    result = g.send(result)
  File "/opt/syntest/synapse/synapse/storage/data_stores/main/events_worker.py", line 155, in get_event
    raise NotFoundError("Could not find event %s" % (event_id,))
synapse.api.errors.NotFoundError: 404: Could not find event $9Ep67Yo6EdH9NpHQaBDkE4Czl_mMf7rqJua7mYkkoAA
```
### Version information

 **Version**: 
 "python_version": "3.7.3",
 "server_version": "1.6.0 (b=develop,70c0da4d8)"

- **Install method**:
  git clone
- **Platform**:
  Debian 10 on ARM64

**Comment by richvdh** [comment]
we see these pretty frequently on matrix.org too. It would be good to figure out what is causing them.

**Comment by richvdh** [comment]
they appear to be due to rejected events. I think, other than the log spam, they are fairly harmless.

**Comment by richvdh** [comment]
I think this got fixed by #6645.
